Framework/PythonInterface/mantid/FunctionWrapper.py

```diff
@@ -0,0 +1,258 @@
+    # get composite parameter name of parameter of 
+    # given name of member function of given index
+        return "f"+str(index)+"."+name
+
+    def __getitem__ (self, nameorindex):
+    # get function of specified index or parameter of specified name
+        item = self.fun.__getitem__(nameorindex)
+        if( isinstance(item, float)):
+           return  item
+        elif( hasattr(item, 'nFunctions')):
+           # item is a CompositeFunction
+           return CompositeFunctionWrapper(item)
+        else:           
+           return FunctionWrapper(item)
+
+    def __setitem__ (self, name, newValue):
+    # set parameter of specified name
+        self.fun.__setitem__(name, newValue)
+                    
+    def __iadd__ (self, other):
+       self.fun.add(other.fun)
+       return self
+       
+    def __delitem__ (self, index):
+       self.fun.__delitem__(index)
+       
+    def __len__ (self):
+       return self.fun.__len__()
+       
+    def __iter__ (self):
+       return self
+       
+    def tieAll (self, name):
+    # For each member function, tie the parameter of the given name 
+    # to the parameter of that name in the first member function.
+    # The named parameter must occur in all the member functions.
+       expr = self.getCompositeParameterName(name, 0)
+       self.tie({self.getCompositeParameterName(name, i): expr for i in range(1,self.__len__()) }) 
+          
+    def fixAll (self, name):
+    # Fix all parameters with the given local name.
+    # Every member function must have a parameter of this name.
+       for i in range(0, self.__len__()):
+          self[i].fix(name)
+          
+    def constrainAll (self, expressions):
+    # Constrain all parameters according local names in expressions.
+       for i in range(0, self.__len__()):
+          if( isinstance( self[i], CompositeFunctionWrapper )):
+             self[i].constrainAll(expressions)
+          else:
+             try:
+                self[i].constrain(expressions)
+             except:
+                pass
+          
+    def unconstrainAll (self, name):
+    # Constrain all parameters according local names in expressions.
+       for i in range(0, self.__len__()):
+          if( isinstance( self[i], CompositeFunctionWrapper )):
+             self[i].unconstrainAll(name)
+          else:
+             try:
+                self[i].unconstrain(name)
+             except:
+                pass
+          
+    def untieAll (self, name):
+    # Untie all parameters with the given local name.
+    # Every member function must have a parameter of this name.
+       for i in range(0, self.__len__()):
+          self[i].untie(name)
+
+    def flatten (self):
+    # Return composite function, equal to self, but with 
+    # every composite function within replaced by
+    # its list of functions, so having a pure list of functions.
+    # This would enable sensible __add__ and __mult__ operations.
```

Comment:
I've tested this with evaluation you recently showed me and find it although it works with just addition and multiplication, it does not give the correct values for multiplication combined with addition.